src/transactionsFactories/delegationTransactionsFactory.ts

```diff
@@ -73,7 +77,12 @@ export class DelegationTransactionsFactory {
         const dataParts = ["addNodes"];
 
         for (let i = 0; i < numNodes; i++) {
-            dataParts.push(...[options.publicKeys[i].hex(), byteArrayToHex(options.signedMessages[i])]);
+            dataParts.push(
+                ...[
+                    options.publicKeys[i].hex(),
+                    this.argSerializer.valuesToStrings([new BytesValue(Buffer.from(options.signedMessages[i]))])[0],
```

Comment:
We should call `valuesToStrings()` only once per factory method (ideally).

Reply:
Now it's only used once per method.